### ISSUE TITLE
es2015-modules-umd fails when using globally installed babel: "We found a path that isn't a NodePath instance"

### ISSUE THREAD

**Issue by andy-hanson** [babel/babel#2644]
I have a problem when I use my globally-installed version of babel to compile a UMD module.
Here's my package.json:

	{
		"scripts": {
			"build": "babel src --out-dir dist",
			"build-fail": "~/.nvm/versions/node/v5.0.0/bin/babel src --out-dir dist"
		},
		"devDependencies": {
			"babel-cli": "6.0.2",
			"babel-plugin-transform-es2015-modules-umd": "6.0.2"
		},
		"babel": {
			"plugins": ["transform-es2015-modules-umd"]
		}
	}

My `src` consists of a single file `foo.js` with the content:

	const x = 1

(The problem does not happen for a completely empty file.)

When I run `npm run build-fail`, this happens:

	> @ build-fail /home/andy/babel-bug
	> ~/.nvm/versions/node/v5.0.0/bin/babel src --out-dir dist

	Error: src/foo.js: We found a path that isn't a NodePath instance
		at Function.get (/home/andy/babel-bug/node_modules/babel-traverse/lib/path/index.js:103:13)
		at TraversalContext.create (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:73:30)
		at TraversalContext.visitMultiple (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:102:25)
		at TraversalContext.visit (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:193:19)
		at Function.traverse.node (/home/andy/babel-bug/node_modules/babel-traverse/lib/index.js:139:17)
		at NodePath.visit (/home/andy/babel-bug/node_modules/babel-traverse/lib/path/context.js:106:22)
		at TraversalContext.visitQueue (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:151:16)
		at TraversalContext.visitQueue (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:157:21)
		at TraversalContext.visitMultiple (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:106:17)
		at TraversalContext.visit (/home/andy/babel-bug/node_modules/babel-traverse/lib/context.js:193:19)

When I run `npm run build`, it works just fine.

My `babel --version` is `6.0.2 (babel-core 6.0.2)` -- both globally and within my package. `which babel` is `/home/andy/.nvm/versions/node/v5.0.0/bin/babel`.

**Comment by sebmck** [comment]
Can you trying using npm 3 and running `npm install` or run `npm dedupe` with npm 2?

**Comment by andy-hanson** [comment]
My `npm --version` is `3.3.6` since I installed node 5.0.0 first.
I tried testing it with node 4.2.1 (which usese npm 2), but `~/.nvm/versions/node/v4.2.1/bin/babel src --out-dir dist` fails also.
I tried `rm -r node_modules; npm install` as well as `npm uninstall -g babel-cli; npm install -g babel-cli` but those didn't change anything.

**Comment by sebmck** [comment]
What's your output of `npm ls`? I think there are multiple versions of `babel-traverse` trying to work on the same tree.

**Comment by andy-hanson** [comment]
├─┬ babel-cli@6.0.2
	│ ├─┬ babel-core@6.0.12
	│ │ ├─┬ babel-code-frame@6.0.12
	│ │ │ ├─┬ chalk@1.1.1
	│ │ │ │ ├── ansi-styles@2.1.0
	│ │ │ │ ├── escape-string-regexp@1.0.3
	│ │ │ │ ├─┬ has-ansi@2.0.0
	│ │ │ │ │ └── ansi-regex@2.0.0
	│ │ │ │ ├── strip-ansi@3.0.0
	│ │ │ │ └── supports-color@2.0.0
	│ │ │ ├── js-tokens@1.0.2
	│ │ │ ├─┬ line-numbers@0.2.0
	│ │ │ │ └── left-pad@0.0.3
	│ │ │ └─┬ repeating@1.1.3
	│ │ │   └─┬ is-finite@1.0.1
	│ │ │     └── number-is-nan@1.0.0
	│ │ ├─┬ babel-generator@6.0.12
	│ │ │ ├─┬ detect-indent@3.0.1
	│ │ │ │ ├── get-stdin@4.0.1
	│ │ │ │ └── minimist@1.2.0
	│ │ │ ├── is-integer@1.0.6
	│ │ │ ├── source-map@0.4.4
	│ │ │ └── trim-right@1.0.1
	│ │ ├── babel-helpers@6.0.12
	│ │ ├── babel-messages@6.0.12
	│ │ ├─┬ babel-traverse@6.0.2
	│ │ │ ├── globals@8.11.0
	│ │ │ └─┬ invariant@2.1.2
	│ │ │   └── loose-envify@1.1.0
	│ │ ├─┬ babel-types@6.0.13
	│ │ │ └── to-fast-properties@1.0.1
	│ │ ├── babylon@6.0.2
	│ │ ├─┬ debug@2.2.0
	│ │ │ └── ms@0.7.1
	│ │ ├── esutils@2.0.2
	│ │ ├─┬ home-or-tmp@1.0.0
	│ │ │ └── os-tmpdir@1.0.1
	│ │ ├── json5@0.4.0
	│ │ ├─┬ minimatch@2.0.10
	│ │ │ └─┬ brace-expansion@1.1.1
	│ │ │   ├── balanced-match@0.2.1
	│ │ │   └── concat-map@0.0.1
	│ │ ├── private@0.1.6
	│ │ ├─┬ regenerator@0.8.35
	│ │ │ ├─┬ commoner@0.10.3
	│ │ │ │ ├── commander@2.5.1
	│ │ │ │ ├─┬ glob@4.2.2
	│ │ │ │ │ └─┬ minimatch@1.0.0
	│ │ │ │ │   ├── lru-cache@2.7.0
	│ │ │ │ │   └── sigmund@1.0.1
	│ │ │ │ ├── graceful-fs@3.0.8
	│ │ │ │ ├── iconv-lite@0.4.13
	│ │ │ │ ├── install@0.1.8
	│ │ │ │ └── q@1.1.2
	│ │ │ ├─┬ defs@1.1.1
	│ │ │ │ ├─┬ alter@0.2.0
	│ │ │ │ │ └── stable@0.1.5
	│ │ │ │ ├── ast-traverse@0.1.1
	│ │ │ │ ├── breakable@1.0.0
	│ │ │ │ ├── esprima-fb@15001.1001.0-dev-harmony-fb
	│ │ │ │ ├── simple-fmt@0.1.0
	│ │ │ │ ├── simple-is@0.2.0
	│ │ │ │ ├── stringmap@0.2.2
	│ │ │ │ ├── stringset@0.2.1
	│ │ │ │ ├── tryor@0.1.2
	│ │ │ │ └─┬ yargs@3.27.0
	│ │ │ │   ├── camelcase@1.2.1
	│ │ │ │   ├─┬ cliui@2.1.0
	│ │ │ │   │ ├─┬ center-align@0.1.2
	│ │ │ │   │ │ └─┬ align-text@0.1.3
	│ │ │ │   │ │   ├─┬ kind-of@2.0.1
	│ │ │ │   │ │   │ └── is-buffer@1.1.0
	│ │ │ │   │ │   └── longest@1.0.1
	│ │ │ │   │ ├── right-align@0.1.3
	│ │ │ │   │ └── wordwrap@0.0.2
	│ │ │ │   ├── decamelize@1.1.1
	│ │ │ │   ├─┬ os-locale@1.4.0
	│ │ │ │   │ └─┬ lcid@1.0.0
	│ │ │ │   │   └── invert-kv@1.0.0
	│ │ │ │   ├── window-size@0.1.2
	│ │ │ │   └── y18n@3.2.0
	│ │ │ ├── esprima-fb@15001.1.0-dev-harmony-fb
	│ │ │ ├─┬ recast@0.10.24
	│ │ │ │ ├── ast-types@0.8.5
	│ │ │ │ └── source-map@0.4.4
	│ │ │ └── through@2.3.8
	│ │ ├── shebang-regex@1.0.0
	│ │ ├─┬ source-map@0.4.4
	│ │ │ └── amdefine@1.0.0
	│ │ └─┬ source-map-support@0.2.10
	│ │   └── source-map@0.1.32
	│ ├─┬ babel-polyfill@6.0.2
	│ │ ├── core-js@1.2.3
	│ │ └─┬ regenerator@0.8.41
	│ │   ├── esprima-fb@15001.1001.0-dev-harmony-fb
	│ │   └─┬ recast@0.10.33
	│ │     └── ast-types@0.8.12
	│ ├── babel-runtime@6.0.12
	│ ├─┬ chokidar@1.2.0
	│ │ ├─┬ anymatch@1.3.0
	│ │ │ └─┬ micromatch@2.2.0
	│ │ │   ├─┬ arr-diff@1.1.0
	│ │ │   │ ├── arr-flatten@1.0.1
	│ │ │   │ └── array-slice@0.2.3
	│ │ │   ├── array-unique@0.2.1
	│ │ │   ├─┬ braces@1.8.2
	│ │ │   │ ├─┬ expand-range@1.8.1
	│ │ │   │ │ └─┬ fill-range@2.2.2
	│ │ │   │ │   ├── is-number@1.1.2
	│ │ │   │ │   ├─┬ randomatic@1.1.0
	│ │ │   │ │   │ └── kind-of@1.1.0
	│ │ │   │ │   └── repeat-string@1.5.2
	│ │ │   │ ├── lazy-cache@0.2.4
	│ │ │   │ ├── preserve@0.2.0
	│ │ │   │ └── repeat-element@1.1.2
	│ │ │   ├── expand-brackets@0.1.4
	│ │ │   ├─┬ extglob@0.3.1
	│ │ │   │ ├─┬ ansi-green@0.1.1
	│ │ │   │ │ └── ansi-wrap@0.1.0
	│ │ │   │ └── success-symbol@0.1.0
	│ │ │   ├── filename-regex@2.0.0
	│ │ │   ├── is-glob@1.1.3
	│ │ │   ├── kind-of@1.1.0
	│ │ │   ├─┬ object.omit@1.1.0
	│ │ │   │ ├─┬ for-own@0.1.3
	│ │ │   │ │ └── for-in@0.1.4
	│ │ │   │ └── isobject@1.0.2
	│ │ │   ├─┬ parse-glob@3.0.4
	│ │ │   │ ├── glob-base@0.3.0
	│ │ │   │ └── is-dotfile@1.0.2
	│ │ │   └─┬ regex-cache@0.4.2
	│ │ │     ├── is-equal-shallow@0.1.3
	│ │ │     └── is-primitive@2.0.0
	│ │ ├── arrify@1.0.0
	│ │ ├── async-each@0.1.6
	│ │ ├── UNMET OPTIONAL DEPENDENCY fsevents@^1.0.0
	│ │ ├── glob-parent@2.0.0
	│ │ ├─┬ is-binary-path@1.0.1
	│ │ │ └── binary-extensions@1.3.1
	│ │ ├─┬ is-glob@2.0.1
	│ │ │ └── is-extglob@1.0.0
	│ │ ├─┬ lodash.flatten@3.0.2
	│ │ │ ├─┬ lodash._baseflatten@3.1.4
	│ │ │ │ ├── lodash.isarguments@3.0.4
	│ │ │ │ └── lodash.isarray@3.0.4
	│ │ │ └── lodash._isiterateecall@3.0.9
	│ │ └─┬ readdirp@2.0.0
	│ │   ├── graceful-fs@4.1.2
	│ │   └─┬ readable-stream@2.0.3
	│ │     ├── core-util-is@1.0.1
	│ │     ├── isarray@0.0.1
	│ │     ├── process-nextick-args@1.0.3
	│ │     ├── string_decoder@0.10.31
	│ │     └── util-deprecate@1.0.2
	│ ├─┬ commander@2.9.0
	│ │ └── graceful-readlink@1.0.1
	│ ├── convert-source-map@1.1.1
	│ ├── fs-readdir-recursive@0.1.2
	│ ├─┬ glob@5.0.15
	│ │ ├─┬ inflight@1.0.4
	│ │ │ └── wrappy@1.0.1
	│ │ ├── inherits@2.0.1
	│ │ └── once@1.3.2
	│ ├── lodash@3.10.1
	│ ├─┬ output-file-sync@1.1.1
	│ │ ├─┬ mkdirp@0.5.1
	│ │ │ └── minimist@0.0.8
	│ │ └── xtend@4.0.0
	│ ├── path-exists@1.0.0
	│ ├── path-is-absolute@1.0.0
	│ ├── slash@1.0.0
	│ ├── source-map@0.5.3
	│ └─┬ v8flags@2.0.10
	│   └── user-home@1.1.1
	└─┬ babel-plugin-transform-es2015-modules-umd@6.0.2
	  ├─┬ babel-plugin-transform-es2015-modules-amd@6.0.2
	  │ └─┬ babel-plugin-transform-es2015-modules-commonjs@6.0.12
	  │   └── babel-plugin-transform-strict-mode@6.0.2
	  └── babel-template@6.0.12

**Comment by andy-hanson** [comment]
This looks like an npm bug. See https://github.com/npm/npm/issues/10168.

**Comment by zowers** [comment]
so what's the solution/workaround?

**Comment by zowers** [comment]
installing `babel-traverse` separately before all other modules changes error
```
TypeError: example.es6: Property object of MemberExpression expected node to be of a type ["Expression"] but instead got undefined
    at Object.validate (web\node_modules\babel-traverse\node_modules\babel-types\lib\definitions\index.js:97:13)
    at Object.validate (web\node_modules\babel-traverse\node_modules\babel-types\lib\index.js:267:9)
    at NodePath._replaceWith (web\node_modules\babel-traverse\lib\path\replacement.js:203:7)
    at NodePath.replaceWith (web\node_modules\babel-traverse\lib\path\replacement.js:181:8)
    at NodePath.replaceInline (web\node_modules\babel-traverse\lib\path\replacement.js:291:17)
    at Array.enter (web\node_modules\babel-plugin-transform-es2015-modules-amd\node_modules\babel-template\lib\index.js:117:12
)
    at NodePath._call (web\node_modules\babel-traverse\lib\path\context.js:72:18)
    at NodePath.call (web\node_modules\babel-traverse\lib\path\context.js:40:14)
    at NodePath.visit (web\node_modules\babel-traverse\lib\path\context.js:102:12)
    at TraversalContext.visitQueue (web\node_modules\babel-traverse\lib\context.js:151:16)
```

**Comment by zowers** [comment]
same issue with `transform-es2015-modules-amd`

**Comment by Awk34** [comment]
I've got the same error with the AMD transformer. Trying to investigate...

**Comment by hzoo** [comment]
@Awk34 the last issue you are getting is the same as https://github.com/babel/babel/issues/2679

**Comment by Awk34** [comment]
@hzoo ah, I see. So should changing my
```js
export default {
    thing
}
```
to
```js
export {
    thing
}
```
fix all that?

**Comment by kyrylkov** [comment]
After upgrading to Babel 6, we have this error with regular Node.js modules

**Comment by brandonhorst** [comment]
I think this is a bug, but manually triggering `npm dedupe` fixed it for me.

```
$ node -v; npm -v                                                                                                                     master
v5.0.0
3.3.6
```

**Comment by zowers** [comment]
dedupe didn't help me

**Comment by d4rkr00t** [comment]
@zowers Can you try just remove node_modules and reinstall them all again? This fixed it for me.

**Comment by sebmck** [comment]
I doubt there's a way we can fix this without recommending npm dedupe. The tree has been tainted and there's no recovering from that.

**Comment by zowers** [comment]
installing nodejs 5 and doing clean `npm install` fixes the issue

**Comment by zowers** [comment]
does not work in node 4.2.1 (LTS) even with dedupe

**Comment by zowers** [comment]
some of the modules depend on babel-runtime@5.0.0 after commit https://github.com/babel/babel/commit/255288cda5cb887eb15e5c2a32eb056bbcb71b0c, maybe this breaks it?

**Comment by vmakhaev** [comment]
Have same error in different place.
```
 amelisa git:(master) ✗ node -v && npm -v
v0.12.7
2.7.5
```

```
/Users/vmakhaev/projects/amelisa/node_modules/babel-core/lib/transformation/file/index.js:519
      throw err;
            ^
Error: /Users/vmakhaev/projects/amelisa/lib/Doc.js: We found a path that isn't a NodePath instance
    at Function.get (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/path/index.js:103:13)
    at TraversalContext.create (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:73:30)
    at TraversalContext.visitMultiple (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:102:25)
    at TraversalContext.visit (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/context.js:193:19)
    at Function.traverse.node (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:139:17)
    at traverse (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:61:12)
    at Function.traverse.hasType (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/node_modules/babel-traverse/lib/index.js:188:3)
    at BlockScoping.wrapClosure (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:491:47)
    at BlockScoping.run (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:383:12)
    at PluginPass.Loop (/Users/vmakhaev/projects/amelisa/node_modules/babel-preset-es2015/node_modules/babel-plugin-transform-es2015-block-scoping/lib/index.js:89:36)
    at newFn (/Users/vmakhaev/projects/amelisa/node_modules/babel-core/node_modules/babel-traverse/lib/visitors.js:233:27)
```

**Comment by schickling** [comment]
+1

**Comment by valeriansaliou** [comment]
Also encountering this issue.

Looks like a bug in the parser at some level:

**Breaks with NodePath message:**

```javascript
return reject(
  `User ${_cur_user_id} disabled email notifications`
);
```

**Does not break and works fine:**

```javascript
return reject("User disabled email notifications");
```

**Comment by sebmck** [comment]
"Fixed" via 24f5ef50b041a8659c904f0e8bdf88d5d0d77448. See the commit message and code comment for an explanation. Deduping is always going to be the most optimal fix.
